addons/project/views/project_views.xml

```diff
@@ -617,8 +617,11 @@
                     <field name="recurrence_id" invisible="1" />
                     <header>
                         <button name="action_assign_to_me" string="Assign to Me" type="object" class="oe_highlight"
-                            attrs="{'invisible' : [('user_id', '!=', False)]}"/>
-                        <field name="stage_id" widget="statusbar" options="{'clickable': '1', 'fold_field': 'fold'}" attrs="{'invisible': [('project_id', '=', False)]}"/>
+                            attrs="{'invisible' : [('user_id', '!=', False)]}" groups="project.group_project_user"/>
```

Comment:
Can you update your commit according to your changes?

Since now with your changes the fields will be still visible for the internal user without any access to project app but they will be readonly. 🙂

Reply:
hello @xavierbol  ,
i have done the changes,
Thank you for the suggestion